openmeetings-util/src/main/java/org/apache/openmeetings/util/OmFileHelper.java

```diff
@@ -104,6 +104,7 @@ public static void setOmHome(File home) {
 		if (Strings.isEmpty(dataDir)) {
 			DATA_HOME = new File(OM_HOME, DATA_DIR);
 		} else {
+			log.info("Overwrite custom DATA_DIR: {}", dataDir);
```

Comment:
Would be great to also move the variable DATA_DIR into applicationContext.xml

Reply:
@solomax we could do that actually. Init DATA_DIR in Application.java, which is a Spring managed class. 

And then when it calls OmFileHlper.setOmHome(File home, String dataHome)

That would be actually quite trivial. And I think better than via env variables.

Reply:
I would move this `log.info` out of `if/else` let it be logged :)))
What is wrong with `ENV` variable?

currently you can set the `DATA_DIR` in your `auto start script` and it will work forever
In case of `applicationContext.xml` it has to be patched after each upgrade ....

Reply:
I would move this log.info out of if/else let it be logged :)))
=> Done

Reply:
What is wrong with ENV variable?
 *  It would be better to have a central/single way of configuring, we splitting some in XML files others in System.getProperty
 * If you run multiple openmeetings webapps in same container you can't configure individual params
 * Env variables are more for usage in shell scripts
 * Env variables make it hard to create unit and integration tests that run on automated environments

Some of those :) :( :)

Reply:
> It would be better to have a central/single way of configuring, we splitting some in XML files others in System.getProperty

`applicationContext` is for spring config only, additionally we have 

1. `env.sh` for Tomcat (optional)
2. conf/server.xml for Tomcat
3. cxf-servlet.xml for REST/SOAP
4. web.xml for webapp fine tuning
5. configuration DB table for configs
6. XML files for languages
7. .........  i believe many more  ............

> If you run multiple openmeetings webapps in same container you can't configure individual params

Is it hypothetical multiple OM instances in the same Tomcat? AFAIK it was never working solution ....

> Env variables are more for usage in shell scripts

it can be used everywhere :)
let it live in shell scripts :))

> Env variables make it hard to create unit and integration tests that run on automated environments

Currently it is being used in unit tests ....